src/scheduler/job_info.c

```diff
@@ -3959,6 +3959,10 @@ create_subjob_from_array(resource_resv *array, int index, char *subjob_name)
 
 	subjob = dup_resource_resv(array, array->server, array->job->queue, err);
 
+	/* make a copy of dependent jobs */
+	subjob->job->depend_job_str = string_dup(array->job->depend_job_str);
+	subjob->job->dependent_jobs = (resource_resv **) dup_array(array->job->dependent_jobs);
```

Comment:
I didn't think subjobs could have dependencies, is this required for the arrayjob?

Reply:
Well, actually this is a bug fix for an issue in runone dependency. 
What happens is that when the scheduler decides to run a subjob, it should ideally mark all the runone dependent jobs (dependent on its parent) as can_not_run. But while creating the subjobs inside the scheduler we were not copying the list of dependent jobs from the parent because of which we were never marking the dependent jobs as not run and end up running more than one job having runone dependency on each other.

Reply:
Ah ok, makes sense.